sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -614,12 +615,12 @@ private[parquet] class ParquetRowConverter(
   }
 
   private trait RepeatedConverter {
-    private var currentArray: ArrayBuffer[Any] = _
+    private[this] val currentArray = new java.util.ArrayList[Any]()
```

Comment:
@JoshRosen, sorry if I'm ignorant about this but why do we need to change `ArrayBuffer` -> `ArrayList`? Seems `ArrayBuffer` itself is mutable and can `clear()` too.

Reply:
From prior experience I've found `ArrayList` to be marginally faster; I ran some quick-and-dirty non-Spark microbenchmarks and this is indeed still the case, but the gain is pretty marginal compared to other factors.

In the interests of code simplicity and clarity, I've backed out that part of the change: the code now uses and `clear()`s a `mutable.ArrayBuffer`: 6d16f596ef6af9fd8946a062f79d0eeace9e1959